src/LuaTransformer.ts

```diff
@@ -4619,8 +4620,12 @@ export class LuaTransformer {
         }
     }
 
-    protected transformLuaTableProperty(node: ts.PropertyAccessExpression): tstl.UnaryExpression {
+    protected transformLuaTableProperty(node: ts.PropertyAccessExpression): tstl.Expression {
         switch (node.name.text) {
+            case "set":
+                return this.transformExpression(node.expression);
```

Comment:
Okay, took me some time to understand how it works and I don't like how transformation flows there. In general, property access isn't equivalent to it's root node, so returning it there isn't correct.

Maybe something like `const [luaTable, methodName] = this.parseLuaTableExpression(expression.expression);`?